src/annotator/test/frame-observer-test.js

```diff
@@ -1,9 +1,34 @@
-import { delay } from '../../test-util/wait';
-import { FrameObserver, onDocumentReady, $imports } from '../frame-observer';
+import { delay, waitFor } from '../../test-util/wait';
+import {
+  FrameObserver,
+  onDocumentReady,
+  onNextDocumentReady,
+  $imports,
+} from '../frame-observer';
+
+function waitForEvent(target, event) {
+  return new Promise(resolve => {
+    target.addEventListener(event, () => resolve());
+  });
+}
+
+function waitForCall(spy) {
+  return waitFor(() => spy.called, 300 /* timeout */);
```

Comment:
This timeout is arbitrary. The default used by `waitFor` is 10ms, which is too short for these tests.

Reply:
This little wrapper/util test function is handy. I wonder if we should add it to the `test-util/wait` module to encourage us to use a consistent mechanism for such things?

Reply:
Makes sense. I'll leave this for a follow-up as this code is repeated in a few other places that could do with refactoring. Also, it might be possible to find a more efficient implementation than polling.